lib/vfscore/main.c

```diff
@@ -1643,6 +1606,46 @@ UK_SYSCALL_R_DEFINE(int, lstat, const char*, pathname, struct stat*, st)
 	return __lxstat(1, pathname, st);
 }
 
+static int __fxstatat_helper(int ver __unused, int dirfd, const char *pathname,
+		struct stat *st, int flags)
+{
+	if (pathname[0] == '/' || dirfd == AT_FDCWD) {
+		return uk_syscall_r_stat(pathname, st);
+	}
+	// If AT_EMPTY_PATH and pathname is an empty string, fstatat() operates on
+	// dirfd itself, and in that case it doesn't have to be a directory.
+	if ((flags & AT_EMPTY_PATH) && !pathname[0]) {
+		return uk_syscall_r_fstat(dirfd, st);
+	}
+
+	struct vfscore_file *fp;
+	int error = fget(dirfd, &fp);
+	if (error) {
+		errno = error;
+		return -1;
+	}
```

Comment:
```suggestion
	if (error)
		return -error;
```

For consistency, `errno` should be used. And the return value should be a negative number.